cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -398,29 +401,46 @@ var (
 )
 
 func isCoreDNSVersionSupported(client clientset.Interface) (bool, error) {
-	isValidVersion := true
-	coreDNSPodList, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(
-		context.TODO(),
-		metav1.ListOptions{
-			LabelSelector: "k8s-app=kube-dns",
-		},
-	)
-	if err != nil {
-		return false, errors.Wrap(err, "unable to list CoreDNS pods")
-	}
+	passed := true
```

Comment:
this can be an integer.

on each poll loop it is reset to zero.
on each valid image ID it is incremented with 1.

after the poll, if passed == "total Pods", the validation has passed.

Reply:
In fact, we don’t need to perform + 1, we only need to operate the flag once the conditions are not met, and finally return the flag